drivers/gpu/drm/bridge/cdns-mhdp.c

```diff
@@ -1795,10 +1882,12 @@ static int mhdp_remove(struct platform_device *pdev)
 
 	drm_bridge_remove(&mhdp->bridge);
 
-	ret = cdns_mhdp_set_firmware_active(mhdp, false);
-	if (ret) {
-		dev_err(mhdp->dev, "Failed to de-activate DP\n");
-		return ret;
+	if (mhdp->hw_enabled) {
+		ret = cdns_mhdp_set_firmware_active(mhdp, false);
+		if (ret) {
+			dev_err(mhdp->dev, "Failed to de-activate DP\n");
+			return ret;
+		}
```

Comment:
This could have a comment about the race with the fw loading.

Reply:
Ooops. This commend opened a can of worms. There is hard to solve race with the mhdp_fw_cb() and mhdp_remove(), if we want to make sure mhdp_fw_cb() does not activate FW after the driver removal has proceeded beyond mhdp->hw_enabled check.